File: strategy/ducks/Quack/Quack.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace ducks.Quack
{
	public class DuckQuack : IQuackBehaviuor
	{
		public DuckQuack(StringBuilder stringBuilder) => _stringBuilder = stringBuilder;

		public void Quack() => _stringBuilder?.Append("quack");

		private readonly StringBuilder _stringBuilder;
	}
}

File: strategy_java/src/ducks/strategies/Quack.java
package ducks.strategies;

import ducks.base.IQuackBehaviour;

public class Quack implements IQuackBehaviour {
    public Quack(StringBuilder sb) {
        _sb = sb;
    }

    @Override
    public void Quack() {
        _sb.append("quack");
    }

    private final StringBuilder _sb;
}
